native/explorer/src/expressions.rs

```diff
@@ -388,7 +388,7 @@ pub fn expr_nil_count(expr: ExExpr) -> ExExpr {
 pub fn expr_n_distinct(expr: ExExpr) -> ExExpr {
     let expr: Expr = expr.resource.0.clone();
 
-    ExExpr::new(expr.n_unique().cast(DataType::Int64))
+    ExExpr::new(expr.n_unique())
```

Comment:
won't this return uint32?

Reply:
Yes, it will. And we will cast after the expressions are applied. Do you prefer to keep cast it on both places?

Reply:
(not that cast will be performed twice. Just once at either places).

Reply:
@philss i am worried that something like `mutate new_column: n_distinct(df["foo"])` will return UInt32, or am I wrong?

Reply:
Oh, ignore me. I see what you are doing. No, I think the current version is better. I think with your version this code will fail:

    mutate new_column: -n_distinct(df["foo"])

Reply:
or maybe we need both. but ideally we should always cast at the expression.

Reply:
Ok, so let me change the question: why do we need this patch? why isn't the previous code enough? It feels like something else is leaking UInt32 :)

Reply:
It is to fix the bug with `expr_count` that does not cast correctly (the comment from the issue https://github.com/elixir-nx/explorer/issues/431#issuecomment-1328219874).

Maybe this is not the ideal way of doing it, since I'm trying to fix a problem from Polars.
I can close this and try to reproduce the problem with polars and the "count" expr.

Reply:
Oh, I see. let's see if we can fix that in Polars then?

Reply:
Let's do this.. I'm going to prepare a reproduction example app. In the meanwhile I'm closing this PR :)